thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/PresenterSavior.java

```diff
@@ -155,48 +196,41 @@ private String generatePresenterId(@NonNull final TiPresenter presenter) {
      * doesn't exist
      */
     @Nullable
-    private synchronized ActivityScopedPresenters getScope(final Activity activity) {
-        final ActivityInstanceObserver detector = mActivityInstanceObserver;
-        if (detector == null) {
-            return null;
-        }
+    private synchronized PresenterScope getScope(final Object host) {
+        if (host instanceof Activity) {
+            final ActivityInstanceObserver detector = mActivityInstanceObserver;
+            if (detector == null) {
+                return null;
+            }
 
-        final String scopeId = detector.getActivityId(activity);
-        if (scopeId == null) {
-            return null;
-        }
+            final Activity activity = (Activity) host;
+            final String scopeId = detector.getActivityId(activity);
+            if (scopeId == null) {
+                return null;
+            }
 
-        return mScopes.get(scopeId);
+            return mScopes.get(scopeId);
+        } else {
+            // currently only Activity is supported as host
+            throw new IllegalStateException(
```

Comment:
We have multiple times this exception.
I thinking about either 
1: Moving to its own method
2: (Preffered) Create a HostNotSupportExcpetion() which will be thrown here. (Preconfigured with the message of course 😉)